src/sentry/seer/utils.py

```diff
@@ -142,7 +143,22 @@ def from_raw(cls, project_id: int, raw_similar_issue_data: RawSeerSimilarIssueDa
         point to an existing group.
 
         """
-        similar_issue_data = raw_similar_issue_data
+        # Unfortunately, we have to hardcode this separately from the `RawSeerSimilarIssueData` type
+        # definition because Python has no way to derive it from the type (nor vice-versa)
+        required_keys = {"stacktrace_distance", "message_distance", "should_group"}
+        optional_keys = {"parent_hash", "parent_group_id"}
+        expected_keys = {*required_keys, *optional_keys}
```

Comment:
nit: since these are constants, you could pull them out of the function to be on the class / a constant at top of file.

Reply:
Good idea. Done.